src/utils.py

```diff
@@ -40,13 +50,19 @@ def get_pull_and_commit_by_commit_sha(event_data, repo):
     return None, None
 
 
+def get_last_commit(pull):
+    return list(pull.get_commits())[-1]
+
+
 def get_labels(pull):
     return [label.name for label in pull.get_labels()]
 
 
-def remove_label(pull, label):
-    labels = get_labels(pull=pull)
-    if label in labels:
+def remove_label(pull, label, labels_from_pull=None):
+    if labels_from_pull is None:
```

Comment:
To reduce API calls, I think that this logic should be kept out of this function.
So either a user would have to get the labels in their code or use already fetched labels.